src/controleur/AdherentCompteController.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+class AdherentCompteController extends Controller
+{
+
+    // Classes PDO utilisées
+
+    // Variables utilisées par la vue
+    private Adherent $adherent;
+
+    // Variables utilisées par le controller
+    private string    $requestMethod;
+
+    public function __construct(string $requestMethod)
+    {
+        $this->getAdherent();
```

Comment:
J'enlèverai la fonction getAdherent pour mettre son contenu au début du constructeur. Comme ca on voit directement la condition qui fait sortir du constructeur. Et rajouter die() ou exit() après le redirect au cas où.

```suggestion
               if (isset($_SESSION['current_adherent'])) {
            $this->adherent = $_SESSION['current_adherent'];
        } else {
            Router::redirect('accueil');
            die();
        }
```